### PR TITLE
feat(dh): validate period for wholesale calculations

### PR DESCRIPTION
**Pull request by ManBearTM**
- Unify StartCalculationType with CalculationType
- Add new query and deprecate old
- Add new GetLatestCalculation graphql file
- Validate period for all wholesale calculations

### CODE REVIEW

File: libs/dh/shared/data-access-mocks/src/lib/wholesale.ts [hunk 1]
```diff
@@ -612,21 +610,6 @@ function getGridAreasQuery() {
...
-      data: {
-        __typename: 'Query',
-        latestBalanceFixing: {
-          __typename: 'Calculation',
-          period: { start: periodStart, end: periodEnd },
-        },
-      },
-    });
-  });
-}
```
Comment (dzhavat) [inline libs/dh/shared/data-access-mocks/src/lib/wholesale.ts:628]: suggestion: one out. Remember to add one as well :) Add mock for GetLatestCalculation.

File: libs/dh/globalization/assets-localization/src/assets/i18n/en.json [hunk 2]
```diff
@@ -925,15 +931,45 @@
...
+            "SECOND_CORRECTION_SETTLEMENT": "Enter I WANT TO RERUN SECOND CORRECTION SETTLEMENT to confirm",
+            "THIRD_CORRECTION_SETTLEMENT": "Enter I WANT TO RERUN THIRD CORRECTION SETTLEMENT to confirm"
+          },
+          "validation": {
+            "BALANCE_FIXING": "I WANT TO RERUN BALANCE FIXING",
+            "WHOLESALE_FIXING": "I WANT TO RERUN WHOLESALE FIXING",
+            "FIRST_CORRECTION_SETTLEMENT": "I WANT TO RERUN FIRST CORRECTION SETTLEMENT",
+            "SECOND_CORRECTION_SETTLEMENT": "I WANT TO RERUN SECOND CORRECTION SETTLEMENT",
+            "THIRD_CORRECTION_SETTLEMENT": "I WANT TO RERUN THIRD CORRECTION SETTLEMENT"
+          },
```
Comment (dzhavat) [inline libs/dh/globalization/assets-localization/src/assets/i18n/en.json:972]: praise: next level validation 💪

**Review by dzhavat — COMMENTED** [review]
praise: nice work. Ony one comment :)

File: libs/dh/globalization/assets-localization/src/assets/i18n/en.json (hunk 2 above)
Comment (ManBearTM) [reply libs/dh/globalization/assets-localization/src/assets/i18n/en.json:972]: Good catch!

2e491098c06189586c848c2f282db6fdebc2c33b

**Review by dzhavat — APPROVED** [review]

**Pull request merged**